src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1208,82 +1122,135 @@ public static string GetPSModulePath(PSModulePathScope scope)
+                    ? entry.Slice(0, 3)
+                    : trimmedEntry;
+#endif
+
+                // If after trimming we don't have a path value we ignore it.
+                if (entry.Length == 0)
                 {
-                    currentProcessModulePath += CombineSystemModulePaths(); // += (SharedModulePath + $PSHome\Modules)
+                    continue;
                 }
-                else
+
+                bool add = true;
+                foreach (ReadOnlySpan<char> listEntry in finalModulePath)
                 {
-                    currentProcessModulePath += hklmMachineModulePath; // += EVT.Machine
+                    if (listEntry.SequenceEqual(entry, valueComparer))
+                    {
+                        add = false;
+                        break;
+                    }
                 }
-            }
-            // EVT.Process exists
-            // Now handle the case where the environment variable is already set.
-            else
-            {
-                string personalModulePathToUse = string.IsNullOrEmpty(hkcuUserModulePath) ? personalModulePath : hkcuUserModulePath;
-                string systemModulePathToUse = string.IsNullOrEmpty(hklmMachineModulePath) ? psHomeModulePath : hklmMachineModulePath;
-
-                // Maintain order of the paths, but ahead of any existing paths:
-                // personalModulePath
-                // sharedModulePath
-                // systemModulePath
-
-                int insertIndex = 0;
-
-                currentProcessModulePath = UpdatePath(currentProcessModulePath, personalModulePathToUse, ref insertIndex);
-                currentProcessModulePath = UpdatePath(currentProcessModulePath, sharedModulePath, ref insertIndex);
-                currentProcessModulePath = UpdatePath(currentProcessModulePath, systemModulePathToUse, ref insertIndex);
-            }
-
-            return currentProcessModulePath;
-        }
 
-        private static string UpdatePath(string path, string pathToAdd, ref int insertIndex)
-        {
-            if (!string.IsNullOrEmpty(pathToAdd))
-            {
-                path = AddToPath(path, pathToAdd, insertIndex);
-                insertIndex = path.IndexOf(Path.PathSeparator, PathContainsSubstring(path, pathToAdd));
-                if (insertIndex != -1)
+                if (add)
                 {
-                    // advance past the path separator
-                    insertIndex++;
+#if UNIX
+                    string pathEntry = entry.ToString();
+#else
+                    string pathEntry;
+                    unsafe
+                    {
+                        // Weird code avoids multiple allocations to replace / with \.
+                        // We can pass entry and remove the Unsafe code can be
+                        // removed once project's LangVersion is at 13.0+.
+                        pathEntry = string.Create(
```

Comment:
Now we are on .Net 9 and we can use LangVersion 13.0.

Reply:
Sure, but until that's done this PR still needs to use the old memory reference way.

Reply:
https://github.com/PowerShell/PowerShell/pull/24617

Reply:
> It's currently on 11

There was simply no need to utilize the new language features.

Reply:
Sure and until it's bumped either by #24617 or another PR I need to keep this older "unsafe" method. I didn't want to bump it in this PR as that's not really related to the changes here.

Reply:
@jborean93 #24617 was merge. You can rebase and simplify the code.

Reply:
Appreciate the PR @xtqqczze and for reviewing merging @iSazonov. Unfortunately https://github.com/PowerShell/PowerShell/blob/060cbe34129238d0a0d37735edb66dc9e984c190/PowerShell.Common.props#L143 was missed and that's what the project is using in this case.

Reply:
https://github.com/PowerShell/PowerShell/pull/24619

Reply:
@jborean93 you can now rebase

Reply:
Thanks that did the trick, commit pushed that removes the now unneeded unsafe code.